readme.md

```diff
@@ -876,11 +876,18 @@ namespace Azure.Service.Models
 
 #### Change the implementation of serialization/deserialization method of one particular property
 
-If you want to change the implementation of serialization/deserialization method of one particular property, you could define your own hook methods and assign them to a property using the `CodeGenMemberSerializationHooks` attribute.
+If you want to change the implementation of serialization/deserialization method of one particular property, you could define your own hook methods and assign them to the property by using the `CodeGenMemberSerializationHooks` attribute.
```

Comment:
well it is the backward compatibility consideration, therefore I introduced this flexibility.
We have quite a few usage in the sdk already adding this attribute to a property.

Reply:
yeah, I got the idea behind it.
But I feel like the same attribute on different places with different required fields is kind of confusing.
Is it possible to change the existing attribute we have in the custom code? Is it part of the public API already?

Reply:
And we do not only have this attribute, we have other similar attributes that only applies to property.

Another reason we should allow this attribute to be applied to a property is, we have the scenario that we are adding new properties and want to add it to the serialization. In this scenario, it would be intuitive if this attribute could apply to the property they just added. Same thing happens to the case when we are changing the type of an existing property.